### ISSUE TITLE
Reward and val acc didnt grow after 30 steps

### ISSUE THREAD

**Issue by Luodian** [hiyouga/EasyR1#24]
Hi～Thank you for providing this wonderful repository!

I tried running the implementation but observed some unexpected behavior. While your report indicates a steadily increasing accuracy trend, I did not see continuous improvement beyond 30 steps. I made some adjustments to the configuration, but I am unsure whether the issue stems from these modifications.

Here is my configuration:

```yaml
data:
  train_files: hiyouga/math12k@train
  val_files: hiyouga/math12k@test
  prompt_key: problem
  rollout_batch_size: 32
  max_prompt_length: 2048
  max_response_length: 2048

algorithm:
  adv_estimator: grpo
  kl_coef: 0.0

worker:
  actor:
    global_batch_size: 8
    micro_batch_size_per_device_for_update: 1
    micro_batch_size_per_device_for_experience: 1
    use_kl_loss: true
    kl_loss_coef: 1.0e-3
    kl_loss_type: low_var_kl
    model:
      model_path: Qwen/Qwen2.5-7B-Instruct
      enable_gradient_checkpointing: true
    optim:
      lr: 1.0e-6
    offload:
      param_offload: true
      optimizer_offload: true

  rollout:
    tensor_parallel_size: 2
    gpu_memory_utilization: 0.7
    n: 8
    temperature: 1.0
    enable_chunked_prefill: true

  ref:
    offload:
      param_offload: true

  reward:
    compute_score: math

trainer:
  total_episodes: 15
  logger: ["console", "wandb"]
  project_name: easy_r1
  experiment_name: qwen2_5_7b_math
  n_gpus_per_node: 8
  nnodes: 1
  save_freq: 5
  test_freq: 5

```

My results:

![Image](https://github.com/user-attachments/assets/249e64eb-d689-4465-948b-1c0d6117890c)

Do you have any insights on what might be causing this issue? Could it be related to the configuration changes?

Looking forward to your thoughts!

**Comment by FaltingsA** [comment]
@Luodian  I guess the stable training of GRPO relies on using a larger batch size for gradient backpropagation.  rollout_batch_size  and global_batch_size may be too small？

**Comment by Luodian** [comment]
Let me try try

**Comment by ICE-Bruce** [comment]
Is the step here means epoch?
